compose-table/src/androidTest/java/org/dhis2/composetable/CellTableTest.kt

```diff
@@ -0,0 +1,170 @@
+package org.dhis2.composetable
+
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import org.dhis2.composetable.activity.TableTestActivity
+import org.dhis2.composetable.data.InputRowOption
+import org.dhis2.composetable.data.TableAppScreenOptions
+import org.dhis2.composetable.model.FakeModelType
+import org.junit.Rule
+import org.junit.Test
+
+class CellTableTest {
+
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<TableTestActivity>()
+
+    @Test
+    fun shouldDisplayMandatoryIcon() {
+        tableRobot(composeTestRule) {
+            val fakeModel = initTable(
+                composeTestRule.activity.applicationContext,
+                FakeModelType.MANDATORY_TABLE
+            )
+            val firstId = fakeModel.first().id!!
+            clickOnCell(firstId, 0, 0)
+            assertCellHasMandatoryIcon(firstId, 0, 0)
+        }
+    }
+
+    @Test
+    fun shouldBlockClickAndSetCorrectColorIfNonEditable() {
+        tableRobot(composeTestRule) {
+            val fakeModel = initTableAppScreen(
+                composeTestRule.activity.applicationContext,
+                FakeModelType.MANDATORY_TABLE
+            )
+            val firstId = fakeModel.first().id!!
+            clickOnCell(firstId, 0, 1)
+            assertInputComponentIsHidden()
+            assertCellBlockedCell(firstId, 0, 1)
+        }
+    }
+
+    @Test
+    fun shouldUpdateValueWhenTypingInComponent() {
+        tableRobot(composeTestRule) {
+            val fakeModel = initTableAppScreen(
+                composeTestRule.activity.applicationContext,
+                FakeModelType.MANDATORY_TABLE
+            )
+            val firstId = fakeModel.first().id!!
+            clickOnCell(firstId, 1, 0)
+            typeOnInputComponent("check")
+            assertCellHasText(firstId, 1, 0, "check")
+        }
```

Comment:
@Balcan Do you think we are missing from the jira issue description the following use case:

1) User clicks in on a cell and then when clicking on the next one the value should be saved?

I have added in my PR a "onSavedTable" cell that is filled everytime on saved is triggered.

Reply:
I forgot... I've just added it